src/event_timer.py

```diff
@@ -69,15 +69,21 @@ def ensure_event_timer(self, event_name: str, interval: int, timeout: Optional[i
         Args:
             event_name: Name of the juju event to schedule.
             interval: Number of minutes between emitting each event.
+            timeout: Timeout for each event handle in minutes.
 
         Raises:
             TimerEnableError: Timer cannot be started. Events will be not emitted.
         """
+        if timeout is not None:
+            timeout_in_secs = timeout * 60
+        else:
+            timeout_in_secs = interval * 30
+
         context: EventConfig = {
```

Comment:
A nitpicky comment, the `EventConfig` definition could have attributes docstring so the time units can be determined :)

Reply:
Pleaes do feel free to resolve this for now!

Reply:
I have added the attribute docstring in the refactoring charm_state branch.